lib/led/rgb.js

```diff
@@ -331,6 +336,61 @@ class RGB {
     return this[this.isOn ? "off" : "on"]();
   }
 
+  /**
+   * pulse Fade the RGB in and out in a loop with specified time
+   * @param  {number} duration Time in ms that a fade in/out will elapse
+   * @return {RGB}
+   *
+   * - or -
+   *
+   * @param  {Object} val An Animation() segment config object
+   */
+
+  // rgb pulse function similar to led pulse function
+  // uses state.prev for the current color
+  pulse(duration, callback) {
+    const state = priv.get(this);
+    var currentColor = state.prev;
+    this.stop();
```

Comment:
[Doc](http://johnny-five.io/api/animation/#segment-properties) for segment properties.